File: medium/4014.py
"""
4014. Minimum Total Price After Applying Discounts

https://leetcode.com/problems/minimum-total-price-after-applying-discounts/

Weekly Contest 514
"""


class Solution:
    def minPrice(self, prices: list[int], discounts: list[int]) -> float:
        prices.sort(reverse=True)
        discounts.sort(reverse=True)

        out: float = 0.0

        for idx in range(min(len(prices), len(discounts))):
            out += prices[idx] * (1 - discounts[idx] / 100)

        for jdx in range(idx + 1, len(prices)):
            out += prices[jdx]

        return round(out, 5)

File: medium/4014.rs
// 4014. Minimum Total Price After Applying Discounts
// https://leetcode.com/problems/minimum-total-price-after-applying-discounts/
// Weekly Contest 514

use std::cmp::min;

impl Solution {
    pub fn min_price(prices: Vec<i32>, discounts: Vec<i32>) -> f64 {
        let mut prices: Vec<i32> = prices;
        let mut discounts: Vec<i32> = discounts;

        prices.sort_unstable_by(|a, b| b.cmp(a));
        discounts.sort_unstable_by(|a, b| b.cmp(a));

        let mut out: f64 = 0.0;

        let middle: usize = min(prices.len(), discounts.len());

        for idx in 0..middle {
            out += prices[idx] as f64 * (1.0 - discounts[idx] as f64 / 100.0);
        }

        for jdx in middle..prices.len() {
            out += prices[jdx] as f64;
        }

        out
    }
}
